framework/Metrics/metrics/RepresentativityFactors.py

```diff
@@ -0,0 +1,113 @@
+    """
+    inputSpecification = super(RepresentativityFactors, cls).getInputSpecification()
+    actionTypeInput = InputData.parameterInputFactory("actionType", contentType=InputTypes.StringType)
+    inputSpecification.addSub(actionTypeInput)
+
+    return inputSpecification
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    # Metric.__init__(self)
+    super().__init__()
+    # The type of given analysis
+    self.actionType                      = None
+    # True indicates the metric needs to be able to handle dynamic data
+    self._dynamicHandling = True
+    # True indicates the metric needs to be able to handle pairwise data
+    self._pairwiseHandling = False
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Method that reads the portion of the xml input that belongs to this specialized class
+      and initialize internal parameters
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    paramInput = Metric.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    for child in paramInput.subparts:
+      if child.getName() == "actionType":
+        self.order = child.value
+      else:
+        self.raiseAnError(IOError, "Unknown xml node ", child.getName(), " is provided for metric system")
+
+  def run(self, x, y, weights = None, axis = 0, **kwargs):
+    """
+      This method computes DSS distance between two inputs x and y based on given metric
+      @ In, x, numpy.ndarray, array containing data of x, if 1D array is provided,
+        the array will be reshaped via x.reshape(-1,1), shape (n_samples, ), if 2D
+        array is provided, shape (n_samples, n_time_steps)
+      @ In, y, numpy.ndarray, array containing data of y, if 1D array is provided,
+        the array will be reshaped via y.reshape(-1,1), shape (n_samples, ), if 2D
+        array is provided, shape (n_samples, n_time_steps)
+      @ In, weights, array_like (numpy.array or list), optional, weights associated
+        with input, shape (n_samples) if axis = 0, otherwise shape (n_time_steps)
+      @ In, axis, integer, optional, axis along which a metric is performed, default is 0,
+        i.e. the metric will performed along the first dimension (the "rows").
+        If metric postprocessor is used, the first dimension is the RAVEN_sample_ID,
+        and the second dimension is the pivotParameter if HistorySet is provided.
+      @ In, kwargs, dict, dictionary of parameters characteristic of each metric
+      @ Out, value, float, metric result
+    """
+    # assert (isinstance(x, np.ndarray))
+    # assert (isinstance(y, np.ndarray))
+    senMeasurables = kwargs['senMeasurables']
+    senFOMs = kwargs['senFOMs']
+    covParameters = kwargs['covParameters']
+    r = (senFOMs.T @ covParameters @ senMeasurables)/\
+        np.sqrt(senFOMs.T @ covParameters @ senFOMs)/\
+        np.sqrt(senMeasurables.T @ covParameters @ senMeasurables)
```

Comment:
Why do you need to create a metric class to do this calculation? It seems to me this calculation is only specific to representativity. How can we use it for other applications? If not, I would suggest to move it the representativity PP.